crates/primitives/src/transaction/sidecar.rs

```diff
@@ -134,7 +134,7 @@ impl BlobTransaction {
 
         // Encode the inner tx list header, then its fields
         tx_header.encode(out);
-        self.transaction.encode_fields(out);
+        self.transaction.encode_with_signature_fields(&self.signature.to_alloy_signature(), out);
```

Comment:
hmm, this now encodes `tx_header` twice

I think we should actually just embed `TxEip4844WithSidecar` into `BlobTransaction`. It already has helper equivalent to this method: https://github.com/alloy-rs/alloy/blob/bfbbc3246bac830dda96d90e9ab759b979f90bdc/crates/consensus/src/transaction/eip4844.rs#L859

However, it is missing some other helpers for length calculation and decoding, we'll likely need to add those to alloy

Reply:
Is it not easier to make `encode_fields` public in alloy?

Reply:
We can add additional helpers later, they wouldn't break existing functions.

I think we want to eventually keep most of rlp logic in alloy anyway, so embedding `TxEip4844WithSidecar` seems to be a reasonable step in scope of this PR which will allow us to delegate 4844 encoding to alloy